app/app/src/main/kotlin/com/hedvig/app/feature/profile/ui/payment/PaymentDestination.kt

```diff
@@ -0,0 +1,321 @@
+    Row(
+      modifier = Modifier
+        .fillMaxWidth()
+        .padding(vertical = 12.dp),
+      horizontalArrangement = Arrangement.SpaceBetween,
+      verticalAlignment = Alignment.CenterVertically,
+    ) {
+      Row(
+        verticalAlignment = Alignment.CenterVertically,
+      ) {
+        Image(
+          painter = painterResource(id = com.hedvig.android.core.ui.R.drawable.ic_pillow),
+          contentDescription = null,
+          modifier = Modifier.size(32.dp),
+        )
+        Spacer(Modifier.width(8.dp))
+        Text(insuranceCost.displayName)
+      }
+      Text(
+        text = insuranceCost.cost.format(locale),
+        color = MaterialTheme.colorScheme.onSurfaceVariant,
+      )
+    }
+    Divider()
+  }
+}
+
+@Composable
+private fun TotalDiscount(uiState: PaymentUiState) {
+  Row(
+    modifier = Modifier
+      .fillMaxWidth()
+      .padding(vertical = 12.dp),
+    horizontalArrangement = Arrangement.SpaceBetween,
+  ) {
+    Text(stringResource(R.string.PAYMENTS_DISCOUNTS_SECTION_TITLE))
+    Text(
+      text = uiState.totalDiscount,
+      color = MaterialTheme.colorScheme.onSurfaceVariant,
+    )
+  }
+}
+
+@Composable
+private fun AddDiscount(uiState: PaymentUiState) {
+  Row(
+    modifier = Modifier.fillMaxWidth(),
+    horizontalArrangement = Arrangement.SpaceBetween,
+    verticalAlignment = Alignment.CenterVertically,
+  ) {
+    Text(stringResource(R.string.PAYMENTS_ADD_CODE_LABEL))
+    Switch(
+      checked = true,
+      onCheckedChange = {},
+      colors = SwitchDefaults.colors(checkedTrackColor = MaterialTheme.colorScheme.typeElement),
+    )
+  }
+  Spacer(Modifier.height(12.dp))
+  uiState.activeDiscounts.forEach {
+    Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
+      Text(it.code)
+      Text(
+        text = it.displayName,
+        color = MaterialTheme.colorScheme.onSurfaceVariant,
+      )
+    }
+    Spacer(Modifier.height(4.dp))
+  }
+  Spacer(Modifier.height(12.dp))
+  Row(
+    modifier = Modifier.fillMaxWidth(),
+    horizontalArrangement = Arrangement.SpaceBetween,
+  ) {
+    HedvigTextField(
+      value = uiState.discountCode ?: "",
+      onValueChange = {},
+      errorText = uiState.discountError,
+      label = {
+        Text(stringResource(id = R.string.REFERRAL_ADDCOUPON_INPUTPLACEHOLDER))
+      },
+      modifier = Modifier.weight(1f),
+    )
+    Spacer(Modifier.width(8.dp))
+    HedvigContainedSmallButton(
+      text = stringResource(id = R.string.PAYMENTS_ADD_CODE_BUTTON_LABEL),
+      onClick = {},
+      contentPadding = PaddingValues(horizontal = 16.dp, vertical = 16.dp),
+      colors = ButtonDefaults.buttonColors(
+        containerColor = MaterialTheme.colorScheme.typeContainer,
+        contentColor = MaterialTheme.colorScheme.primary,
+      ),
+      modifier = Modifier.widthIn(min = 127.dp),
+    )
+  }
+}
+
+@Composable
+fun PaymentDetails(uiState: PaymentUiState) {
```

Comment:
This function takes in the UiState I think only to use one field from it, the display name. Maybe it can just take in that one parameter only?
```suggestion
fun PaymentDetails(paymentMethodDisplayName: String) {
```